solveengine/basemodel.py

```diff
@@ -33,31 +34,32 @@ class BaseModel(object):
     """
     OPTIONS = namedtuple("Options", 'sleeptime debug')
 
-    def __init__(self, token, filename="model", sleeptime=2, debug=False, 
+    def __init__(self, token, filename="model", run_time_limit=600, sleeptime=2, debug=False,
                  file_ending=".lp", interactive_mode=False, http_mode=False):
         if debug:
             LOGGER.setLevel(logging.DEBUG)
         if file_ending not in [".lp", ".cnf"]:
             raise ValueError("Filetype {} not supported".format(file_ending))
-        self._file_ending = file_ending
-        self._filename = filename + file_ending
-        self._token = token
-        self._id = None
-        self._options = BaseModel.OPTIONS(sleeptime, debug)
-        self._solver_status = str(SolverStatusCode.NOTSTARTED)
-        self._se_status = str(SEStatusCode.NOTSTARTED)
-        
+        self.__file_ending = file_ending
+        self.__filename = "".join([filename, self.__file_ending])
+        self.__token = token
+        self.__id = None
+        self.__options = BaseModel.OPTIONS(sleeptime, debug)
+        self.__solver_status = str(SolverStatusCode.NOTSTARTED)
+        self.__se_status = str(SEStatusCode.NOTSTARTED)
+        #self.run_time_limit = run_time_limit
```

Comment:
preparation for later argument : optional timeout
but you are right to spot it, I have actually now removed it